subiquity/server/server.py

```diff
@@ -178,6 +178,20 @@ async def free_only_POST(self, enable: bool) -> None:
         # enabling free only mode means disabling components
         self.app.base_model.mirror.disable_components(to_disable, enable)
 
+    async def interactive_sections_GET(self) -> Optional[List[str]]:
+        if self.app.autoinstall_config is None:
+            return None
+
+        i_sections = self.app.autoinstall_config.get(
+                'interactive-sections', None)
+        if i_sections == ['*']:
+            # expand the asterisk to the actual controller key names
+            return [controller.autoinstall_key
+                    for controller in self.app.controllers.instances
+                    if controller.interactive()]
+
```

Comment:
Argl. Is the autoinstall key really what the desktop team want here? Usually controller_name and autoinstall_key and model_name and the name of the endpoint (and ...) are all the same but they are not always. For example for apt configuration the autoinstall key is "apt" but the endpoint in the api is "/mirror". Which does the client actually want?

Also controller.autoinstall_key will be None for some things (they should be filtered out in this context I think). I think an api test is probably a good idea?

cc @jpnurmi

Reply:
According to https://ubuntu.com/server/docs/install/autoinstall-reference the sections that can be interactive are:

- locale
- refresh-installer
- keyboard
- source
- network
- proxy
- apt
- storage
- identity
- active-directory
- ubuntu-pro
- ssh
- drivers
- snaps

This list looks ok to me. It's pretty close and easily mappable to the page route names we have today.

Some of these sections don't exist in the desktop GUI at all, though, and source & drivers are on the same page, for example. Given that there's no 1:1 mapping anyway, it'd be fine to use either autoinstall keys or model names whichever makes more sense.

Reply:
API test added.  We should expand with autoinstall_key, since that's the correct value for specifying the desired sections.